### PR TITLE
InvokeCommandAction - add additional properties for more MVVM scenarios

### PR DESCRIPTION
**Pull request by brianlagunas**
Added initial EventArgs related properties to the InvokeCommandAction

Properties Added:
- EventArgsConverter: Instance of IValueConverter that converts the EventArgs into a more MVVM friendly value
- EventArgsConverterParameter: the parameter that will be sent as the parameter argument to IValueConverter.Convert method
- EventArgsParameterPath: parameter path to extract a property from the EventArgs that will be passed to ICommand
- PassEventArgsToCommand: forces the EventArgs to be passed to the ICommand if the other properties are null

## EventArgsConverter
Using the EventArgsConverter to retrieve the ItemTappedEventArgs.Item property.
```
public class ItemTappedEventArgsConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            var itemTappedEventArgs = value as ItemTappedEventArgs;
            if (itemTappedEventArgs == null)
            {
                throw new ArgumentException("Expected value to be of type ItemTappedEventArgs", nameof(value));
            }
            return itemTappedEventArgs.Item;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
```
In XAML, add a reference to the converter and the converter resource need to be defined
```
<b:InvokeCommandAction Command="{Binding ItemTappedCommand}"
                        EventArgsConverter="{StaticResource itemTappedEventArgsConverter}" />
```

## EventArgsParameterPath
Sample EventArgs
```
public class ItemTappedEventArgs : EventArgs
{
    public object Item { get; }
    public object Group { get; }
}
```
Setting EventArgsParameterPath to **Item** will extract the property value and pass it to the ICommand
```
<b:InvokeCommandAction Command="{Binding ItemTappedCommand}"
                                         EventArgsParameterPath="Item" />
```
Nested properties are also supported:
```
<b:InvokeCommandAction Command="{Binding ItemTappedCommand}"
                                         EventArgsParameterPath="Item.SubItem.SubProperty" />
```

### CODE REVIEW

File: src/Microsoft.Xaml.Behaviors/InvokeCommandAction.cs [hunk 1]
```diff
@@ -76,27 +112,60 @@ protected override void Invoke(object parameter)
...
         private ICommand ResolveCommand()
         {
             ICommand command = null;
 
             if (this.Command != null)
             {
                 command = this.Command;
-            }
-            else if (this.AssociatedObject != null)
+            } else if (this.AssociatedObject != null)
```
Comment (nigel-sampson) [inline src/Microsoft.Xaml.Behaviors/InvokeCommandAction.cs]: Looking through the codebase it seems the standard is for it to be on the new line. (another instance of this further up the file)

Comment (brianlagunas) [reply src/Microsoft.Xaml.Behaviors/InvokeCommandAction.cs]: Yeah, we have an editorconfig file that we need to update. I don't like this either. We should add an issue for an updated editorconfig file and set the new standards

**Comment by brianlagunas** [comment]
One thing I can do is add more detail to the comments to indicate when the property is used.  If it makes you feel any better, other frameowkrs such as Prism, MVVMLight, MVVMCross, and others have had this for many years and everyone seems to understand how it works, or at least learned how it works :)

**Comment by nigel-sampson** [comment]
Fair enough, it's not a major concern for me.

**Pull request merged**